torch/_inductor/decomposition.py

```diff
@@ -870,12 +906,25 @@ def fast_random_decomps() -> dict[Any, Callable[..., Any]]:
     return {**decompositions, **extra_random_decomps}
 
 
+# Random *_like ops that should be excluded when fallback_random=True
+_random_like_ops = [
+    aten.rand_like,
+    aten.randn_like,
+    aten.randint_like.default,
+    aten.randint_like.low_dtype,
+]
+
+
 # TODO(aakhundov): replace this (and the above) Any by more
 # specific type and fix all the cascading mypy errors
 def select_decomp_table() -> dict[Any, Callable[..., Any]]:
     """decomps can change based on config"""
     if config.fallback_random:
-        return decompositions
+        # Exclude random *_like ops when fallback_random=True so they use native implementations
+        decomps_without_random_like = {
+            k: v for k, v in decompositions.items() if k not in _random_like_ops
```

Comment:
i explored whether to move the _like decomps from this file into torch/_decomps/decompositions_for_rng.py but that didn't look like the right fix because that file seems more to do with functionalization